internal/controller/pipelinerollout_controller.go

```diff
@@ -444,6 +460,7 @@ func (r *PipelineRolloutReconciler) processExistingPipelineWithoutDataLoss(ctx c
 				return err
 			}
 			pipelineRollout.Status.MarkDeployed(pipelineRollout.Generation)
+			r.customMetrics.ReconciliationDuration.WithLabelValues(ControllerPipelineRollout, "update").Observe(time.Since(syncStartTime).Seconds())
```

Comment:
Do we need this one? I think the one you have above in the caller will already take care of this?